posts/inside-rust/2022-09-22-announcing-the-rust-style-team.md

```diff
@@ -0,0 +1,61 @@
+makes it easy to maintain and to incorporate in continuous integration.
+`rustfmt` also provides many options to customize the style, but the style
+guide defines the defaults, and most projects use those defaults.
+
+The standard Rust style resulted from development and discussion within the
+Rust style team, between 2016 and 2018. After publishing the style guide, the
+Rust style team wound up its active work, by design.
+
+However, as the Rust language develops, we have a regular need for improvements
+to the style guide, such as to support new language constructs. This includes
+minor language changes, as well as highly anticipated new features such as
+`let`-chaining (RFC 2497) and `let`-`else` (RFC 3137). New constructs like
+these, by default, get ignored and not formatted by rustfmt, and subsequently
+need formatting added. Some of this work has fallen to the rustfmt team in
+recent years, but the rustfmt team would prefer to implement style
+determinations made by another team rather than making such determinations
+itself.
+
+In addition, rustfmt maintains backwards compatibility guarantees: code that
+has been correctly formatted with rustfmt won't get formatted differently with
+a future version of rustfmt. This avoids churn, and avoids creating CI failures
+when people use rustfmt to check style in CI. However, this also prevents
+evolving the Rust style to take community desires into account and improve
+formatting over time. rustfmt provides various configuration options to change
+its default formatting, and many of those options represent changes that many
+people in the community would like enabled by default.
+
+For instance, many people prefer to format their `use` lines in three blocks:
+imports from the standard library, imports from external crates, and then
+imports from modules within the same project. `rustfmt` supports this via the
+option `group_imports = StdExternalCrate`, but cannot make this the default
+without causing CI failures in existing projects.
+
+To solve both of these problems, [RFC
+3309](https://rust-lang.github.io/rfcs/3309-style-team.html) has revived the
+Rust style team, with three goals:
+
+- Making determinations about styling for new Rust constructs
+- Evolving the existing Rust style
+- Defining mechanisms to evolve the Rust style while taking backwards
+  compatibility into account, such as via Rust editions or similar mechanisms
+
+We don't plan to make any earth-shattering style changes; the look and feel of
+Rust will remain largely the same. Evolutions to the default Rust style will
+largely consist of established `rustfmt` options people already widely enable,
+or would enable if they were stable.
+
+We expect that the initial work of the style team will focus on clearing a
+backlog of new language constructs that lack formatting guidance. Afterwards,
+we look forward to introducing a few minor style evolutions in the Rust 2024
+edition.
```

Comment:
I think in the same vein as @djc's question above, this is giving a little bit of a teaser in terms of some initial thoughts we have relative to the third bullet above about the solution/implementation for the style evolution mechanism. For the purposes of an announcement type of blog post, perhaps it would be better to not tease any specifics around the early thinkign (especially since we've not discussed this with the totality of the new team), and just mention shifting focus to the evolution mechanism.

Maybe something like:

```suggestion
backlog of new language constructs that lack formatting guidance. Afterwards,
we will look towards defining and implementing the mechanisms to evolve the default Rust style which we hope to then use to introduce a few minor style improvements in the near future.
```

Reply:
@calebcartwright I'd meant this to be an indicator of timeframe rather than specific solution, but you're right, this comes across as implying it'll be tied to edition. Will rework.